core/server/src/main/java/alluxio/web/UIWebServer.java

```diff
@@ -67,22 +72,38 @@ public UIWebServer(ServiceType service, InetSocketAddress address, Configuration
     QueuedThreadPool threadPool = new QueuedThreadPool();
     int webThreadCount = mConfiguration.getInt(Constants.WEB_THREAD_COUNT);
 
-    mServer = new Server();
-    SelectChannelConnector connector = new SelectChannelConnector();
-    connector.setHost(address.getHostName());
-    connector.setPort(address.getPort());
-    connector.setAcceptors(webThreadCount);
-    mServer.setConnectors(new Connector[] {connector});
-
     // Jetty needs at least (1 + selectors + acceptors) threads.
     threadPool.setMinThreads(webThreadCount * 2 + 1);
     threadPool.setMaxThreads(webThreadCount * 2 + 100);
-    mServer.setThreadPool(threadPool);
+
+    mServer = new Server(threadPool);
+
+    mServerConnector = new ServerConnector(mServer);
+    mServerConnector.setPort(mAddress.getPort());
+    mServerConnector.setHost(mAddress.getAddress().getHostAddress());
+
+    mServer.addConnector(mServerConnector);
+
+    System.setProperty("org.apache.jasper.compiler.disablejsr199", "false");
 
     mWebAppContext = new WebAppContext();
     mWebAppContext.setContextPath(AlluxioURI.SEPARATOR);
     File warPath = new File(mConfiguration.get(Constants.WEB_RESOURCES));
     mWebAppContext.setWar(warPath.getAbsolutePath());
+
```

Comment:
These were taking from the examples provided by [Jetty](https://www.eclipse.org/jetty/documentation/9.3.x/embedded-examples.html#embedded-webapp-jsp)

I omitted the comments. But they are 

>  // This webapp will use jsps and jstl. We need to enable the
        // AnnotationConfiguration in order to correctly
        // set up the jsp container

and 

>  // Set the ContainerIncludeJarPattern so that jetty examines these
        // container-path jars for tlds, web-fragments etc.
        // If you omit the jar that contains the jstl .tlds, the jsp engine will
        // scan for them instead.